app/components/thumbnail.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+const MAX_THUMBNAIL_DIMENSION = 999;
+
+export default class Thumbnail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      failed: false,
+    };
+  }
+
+  getThumbnailSrc({ origin, width, height, src }) {
+    const srcPath = src.split('//').pop();
+    return (`${origin}/${width}x${height}/${srcPath}`);
+  }
+
+  handleError() {
+    if (!this.state.failed) {
+      this.setState({ failed: true });
+    }
+  }
+
+  render() {
+    const src = this.state.failed ? this.props.src : this.getThumbnailSrc(this.props);
+
+    const dimensions = {
+      width: null,
+      height: null,
+    };
+
+    const style = {
+      maxWidth: this.props.width,
+      maxHeight: this.props.height,
+    };
+
+    if (this.props.format === 'mp4') {
+      return (
+        <div>
+          <video width="300" controls>
+            <source src={this.props.src} type="video/mp4" />
+           </video>
+        </div>
+      );
+    }
+
+    return (
+      <img alt="" {...this.props} src={src} {...dimensions} style={style} onError={this.handleError} />
```

Comment:
It adds a bit of flexibility to the component (e.g. using arbitrary image attributes)